nilearn/decoding/tests/test_decoder.py

```diff
@@ -184,51 +184,91 @@ def test_decoder_binary_classification():
     y_pred = model.predict(X)
     assert accuracy_score(y, y_pred) > 0.95
 
-    # decoder object use prior as strategy (default) for dummy classifier
+    # check different screening_percentile value
+    for screening_percentile in [100, 20, None]:
+        model = Decoder(mask=mask, screening_percentile=screening_percentile)
+        model.fit(X, y)
+        y_pred = model.predict(X)
+        assert accuracy_score(y, y_pred) > 0.95
+
+    for clustering_percentile in [100, 99]:
+        model = FREMClassifier(estimator='logistic_l2', mask=mask,
+                               clustering_percentile=clustering_percentile,
+                               screening_percentile=90, cv=5)
+        model.fit(X, y)
+        y_pred = model.predict(X)
+        assert accuracy_score(y, y_pred) > 0.9
+
+    # check cross-validation scheme and fit attribute with groups enabled
+    rand_local = np.random.RandomState(42)
+    for cv in [KFold(n_splits=5), LeaveOneGroupOut()]:
+        model = Decoder(estimator='svc', mask=mask,
+                        standardize=True, cv=cv)
+        if isinstance(cv, LeaveOneGroupOut):
+            groups = rand_local.binomial(2, 0.3, size=len(y))
+        else:
+            groups = None
+        model.fit(X, y, groups=groups)
+        assert accuracy_score(y, y_pred) > 0.9
+
+
+def test_decoder_dummy_classifier():
+    n_samples = 200
+    X, y = make_classification(n_samples=n_samples, n_features=125, scale=3.0,
+                               n_informative=5, n_classes=2, random_state=42)
+    X, mask = to_niimgs(X, [5, 5, 5])
+
+    # We make 80% of y to have value of 1.0 to check whether the stratified
+    # strategy returns a proportion prediction value of 1.0 of roughly 80%
+    proportion = 0.8
+    y = np.zeros(n_samples)
+    y[:int(proportion * n_samples)] = 1.0
+
     model = Decoder(estimator='dummy_classifier', mask=mask)
     model.fit(X, y)
     y_pred = model.predict(X)
-    assert model.scoring == "roc_auc"
-    assert model.score(X, y) == 0.5
-    assert accuracy_score(y, y_pred) == 0.5
+    assert np.sum(y_pred == 1.0) / n_samples - proportion < 0.5
```

Comment:
Isn't this bound  bit loose ?

Reply:
Yes, potentially so, but because the Dummy Classifier use numpy.random.multinomial to generate prediction label based on the proportion of label in training set so there would be a fair amount of randomness.

I think one way to decrease the bound here and still guarantee the test would not fail due to randomness is to increase the sample size (maybe 3x-4x), wdyt?

Reply:
Yes, if it's not too time consuming.